mongo/src/main/java/org/axonframework/extensions/mongo/eventhandling/deadletter/DeadLetterEntry.java

```diff
@@ -0,0 +1,534 @@
+        this.processingStarted = null;
+    }
+
+    /**
+     * Sets some indexes needed for the dead letter collection
+     *
+     * @param deadLetterCollection the {@link MongoCollection} to create the indexes on
+     */
+    public static void ensureDeadLetterIndexes(@Nonnull MongoCollection<Document> deadLetterCollection) {
+        deadLetterCollection.createIndex(
+                new BasicDBObject(PROCESSING_GROUP_KEY, ORDER_ASC),
+                new IndexOptions().unique(false).name("processingGroupIdentifierIndex"));
+        deadLetterCollection.createIndex(
+                new BasicDBObject(PROCESSING_GROUP_KEY, ORDER_ASC)
+                        .append(SEQUENCE_IDENTIFIER_KEY, ORDER_ASC),
+                new IndexOptions().unique(false).name("processingGroupAndSequenceIdentifierIndex"));
+        deadLetterCollection.createIndex(
+                new BasicDBObject(PROCESSING_GROUP_KEY, ORDER_ASC)
+                        .append(SEQUENCE_IDENTIFIER_KEY, ORDER_ASC)
+                        .append(INDEX_KEY, ORDER_ASC),
+                new IndexOptions().unique(true).name("uniqueDeadLetterEntryIndex"));
+    }
+
+    /**
+     * Filter only on processingGroup
+     *
+     * @param processingGroup the processing group
+     * @return a {@link Bson} filter
+     */
+    public static Bson processingGroupFilter(@Nonnull String processingGroup) {
+        return eq(PROCESSING_GROUP_KEY, processingGroup);
+    }
+
+    /**
+     * Filter on processingGroup and sequenceIdentifier
+     *
+     * @param processingGroup    the processing group
+     * @param sequenceIdentifier the sequence identifier
+     * @return a {@link Bson} filter
+     */
+    public static Bson processingGroupAndSequenceIdentifierFilter(
+            @Nonnull String processingGroup,
+            @Nonnull String sequenceIdentifier
+    ) {
+        return and(eq(PROCESSING_GROUP_KEY, processingGroup),
+                   eq(SEQUENCE_IDENTIFIER_KEY, sequenceIdentifier));
+    }
+
+    /**
+     * Filter on processingGroup, sequenceIdentifier and index
+     *
+     * @param processingGroup    the processing group
+     * @param sequenceIdentifier the sequence identifier
+     * @param index              the index
+     * @return a {@link Bson} filter
+     */
+    public static Bson uniqueFilter(
+            @Nonnull String processingGroup,
+            @Nonnull String sequenceIdentifier,
+            long index
+    ) {
+        return and(eq(PROCESSING_GROUP_KEY, processingGroup),
+                   eq(SEQUENCE_IDENTIFIER_KEY, sequenceIdentifier),
+                   eq(INDEX_KEY, index));
+    }
+
+    /**
+     * Filter on processingGroup, sequenceIdentifier where the index needs to be higher
+     *
+     * @param processingGroup    the processing group
+     * @param sequenceIdentifier the sequence identifier
+     * @param index              the index
+     * @return a {@link Bson} filter
+     */
+    public static Bson indexGreaterThanFilter(
```

Comment:
Similarly, I think this can be named better. `nextItemInSequenceFilter`, for example

Reply:
In addition, what's the reason for these query specifiers to be in this class? I would expect them in the `DLQ` implementation